resolver/resolver.cc

```diff
@@ -2692,6 +2692,7 @@ class ResolveTypeMembersAndFieldsWalk {
             case core::Names::typeAlias().rawId():
             case core::Names::typeMember().rawId():
             case core::Names::typeTemplate().rawId():
+            case core::Names::syntheticTypeMember().rawId():
```

Comment:
The `Sorbet::Private::Static.type_member` suggestion described elsewhere in my review comments would mean fewer special cases for this `syntheticTypeMember` thing, which means fewer chances to introduce bugs in the future from forgetting to deal with this third synthetic name.